src/TS_HW5/hw5.ts

```diff
@@ -0,0 +1,127 @@
+/*  Створіть класи Circle, Rectangle, Square і Triangle.
+    У кожного з них є загальнодоступний метод calculateArea.
+    У кожної фігури є загальнодоступні властивості - колір і назва, які не можна змінювати після створення.
+    У Square і Rectangle зі свого боку є ще додатковий метод print, який виводить рядок із формулою розрахунку площі.
+*/
+enum Colors {
+  Red = 'Red',
+  Orange = 'Orange',
+  Yellow = 'Yellow',
+  Green = 'Green',
+  LigthBlue = 'Ligth Blue',
+  Blue = 'Blue',
+  Violet = 'Violet',
+}
+
+enum Names {
+  Circle = 'Circle',
+  Rectangle = 'Rectangle',
+  Square = 'Square',
+  Triangle = 'Triangle',
+}
+
+abstract class Figure {
+  protected abstract readonly Color: Colors;
+  protected abstract readonly Name: Names;
+  protected abstract CalculateArea(numberA: number, numberB?: number, numberC?: number): number;
+}
+
+class Circle extends Figure {
+  constructor(
+    public readonly Color: Colors,
+    public readonly Name: Names
+  ) {
+    super();
+  }
+
+  public CalculateArea(radius: number): number {
+    if (!(radius > 0)) throw new RangeError('Radius is less or equal to zero!');
+    return Math.PI * Math.pow(radius, 2);
+  }
+}
+
+class Rectangle extends Figure {
+  constructor(
+    public readonly Color: Colors,
+    public readonly Name: Names
+  ) {
+    super();
+  }
+
+  public CalculateArea(dimensionA: number, dimensionB: number): number {
+    if (!Validator.isPositive(dimensionA, dimensionB))
+      throw new RangeError('One or both dimensions less or equal to zero!');
+    return dimensionA * dimensionB;
+  }
+
+  public Print(): string {
```

Comment:
What measures can I take to prevent forgetting the 'Print' method in that class? You already know how to handle this.

Reply:
Added additional abstract class